lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -823,15 +839,75 @@ def upload_reads(self, ctx, params):
         self.log('Starting upload reads, parsing args')
         o, wsid, name, objid, kbtype, single_end, fwdid, revid, shock = (
             self._proc_upload_reads_params(params))
-        interleaved = 1 if (not single_end and not revid) else 0
+        # IF from shock fwdid and revid are shock nodes, if not shock they are file paths
+        dfu = DataFileUtil(self.callback_url)
+        fwdpath = None
+        revpath = None
```

Comment:
It is redundant, but it also doesn't bother me too much here one way or the other.  For readability, the better improvement I think would be to extract that following large IF clause into a separate function that gives returns the paths you need.  But I'm also ok with this as is.